app/js/stats/controllers/StatsCtrl.js

```diff
@@ -0,0 +1,270 @@
+"use strict";
+/**
+ * @module   FM.stats.StatsCtrl
+ * @author   "SOON_"
+ */
+angular.module("FM.stats.StatsCtrl", [
+    "FM.api.StatsResource",
+    "ngRoute",
+    "chart.js",
+    "ui.bootstrap.datepicker"
+])
+/**
+ * @method config
+ * @param  {Provider} $routeProvider
+ */
+.config([
+    "$routeProvider",
+    function ($routeProvider) {
+
+        $routeProvider
+            .when("/stats", {
+                templateUrl: "partials/stats.html",
+                controller: "StatsCtrl",
+                resolve: {
+                    stats: ["StatsResource", "$route", function (StatsResource, $route){
+                        return StatsResource.get($route.current.params).$promise;
+                    }]
+                }
+            });
+
+    }
+])
+/**
+ * @constructor
+ * @param {Object} $scope Application data on scope
+ * @param {Object} stats  Data resolved from API
+ */
+.controller("StatsCtrl", [
+    "$scope",
+    "$q",
+    "$filter",
+    "$location",
+    "CHART_COLOURS",
+    "StatsResource",
+    "stats",
+    function ($scope, $q, $filter, $location, CHART_COLOURS, StatsResource, stats) {
+
+        /**
+         * Current search params
+         * @property {Object} search
+         */
+        $scope.search = $location.search();
+
+        /**
+         * @property {Object} filter
+         */
+        $scope.filter = {};
+
+        /**
+         * Track open status of datepickers
+         * @property {Object} datepickerOpened
+         */
+        $scope.datepickerOpened = {
+            from: "",
+            to: ""
+        };
+
+        /**
+         * Total number of tracks played
+         * @property {Number} totalPlays
+         */
+        $scope.totalPlays = stats.total_plays;  // jshint ignore:line
+
+        /**
+         * Total play time in ms
+         * @property {Number} totalPlayTime
+         */
+        $scope.totalPlayTime = stats.total_play_time;  // jshint ignore:line
+
+        /**
+         * List of top 10 tracks
+         * @property {Array} topTracks
+         */
+        $scope.topTracks = stats.most_played_tracks;  // jshint ignore:line
+
+        /**
+         * List of top 10 artists
+         * @property {Array} topArtists
+         */
+        $scope.topArtists = stats.most_played_artists;  // jshint ignore:line
+
+        /**
+         * List of top 10 genres
+         * @property {Array} topGenres
+         */
+        $scope.topGenres = stats.most_played_genres;  // jshint ignore:line
```

Comment:
would it be simpler and save a few lines of code if we just saves the stats data in `$scope.stats`? that way if more are added then you wouldn't need to bind each one to the $scope.